spec/validators/waste_exemptions_engine/postcode_validator_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module Test
+  PostcodeValidatable = Struct.new(:postcode) do
+    include ActiveModel::Validations
+
+    validates :postcode, "waste_exemptions_engine/postcode": true
+  end
+end
+
+module WasteExemptionsEngine
+  RSpec.describe PostcodeValidator, type: :model do
+    let(:valid_postcode) { "BS1 5AH" }
+    let(:invalid_postcode) { "foo" }
+    let(:postcode_without_addresses) { "AA1 1AA" }
+
+    subject(:validatable) { Test::PostcodeValidatable.new(valid_postcode) }
+
+    describe "#validate_each" do
+      context "when the postcode is valid" do
+        before(:context) { VCR.insert_cassette("postcode_valid") }
+        after(:context) { VCR.eject_cassette }
+
+        it "gets called" do
+          expect_any_instance_of(PostcodeValidator)
+            .to receive(:validate_each)
+            .once
+
+          validatable.valid?
+        end
+
+        it "the errors are empty" do
+          validatable.valid?
+          expect(validatable.errors).to be_empty
+        end
```

Comment:
I think the one additional test I'd like to see is that the object we have added the validator too is valid.

```ruby
it "confirms the object is valid" do
  expect(validatable).to be_valid
end
```

At the moment it feels like we are assuming this from other test results rather than directly checking.

Reply:
Fixed in the latest commit.